sources/tools/Stride.Importer.Assimp/MeshConverter.cs

```diff
@@ -0,0 +1,1364 @@
+                    }
+
+                    for (uint colorChannel = 0; colorChannel < Utils.GetNumColorChannels(mesh); ++colorChannel)
+                    {
+                        var color = mesh->MColors[(int)colorChannel][i].ToStrideColor();
+                        *((Color*)(vbPointer + colorOffset + sizeColor * colorChannel)) = color;
+                    }
+
+                    if (mesh->MTangents != null)
+                    {
+                        var tangentPointer = (Vector3*)(vbPointer + tangentOffset);
+                        var bitangentPointer = (Vector3*)(vbPointer + bitangentOffset);
+                        *tangentPointer = mesh->MTangents[i].ToStrideVector3();
+                        *bitangentPointer = mesh->MBitangents[i].ToStrideVector3();
+                        if (float.IsNaN(tangentPointer->X) || float.IsNaN(tangentPointer->Y) || float.IsNaN(tangentPointer->Z) ||
+                            float.IsNaN(bitangentPointer->X) || float.IsNaN(bitangentPointer->Y) || float.IsNaN(bitangentPointer->Z))
+                        {
+                            //assert(mesh->HasNormals());
+                            var normalPointer = ((Vector3*)(vbPointer + normalOffset));
+                            Vector3 c1 = Vector3.Cross(*normalPointer, new Vector3(0.0f, 0.0f, 1.0f));
+                            Vector3 c2 = Vector3.Cross(*normalPointer, new Vector3(0.0f, 1.0f, 0.0f));
+
+                            if (c1.LengthSquared() > c2.LengthSquared())
+                                *tangentPointer = c1;
+                            else
+                                *tangentPointer = c2;
+                            *bitangentPointer = Vector3.Cross(*normalPointer, *tangentPointer);
+                        }
+                        tangentPointer->Normalize();
+                        bitangentPointer->Normalize();
+                    }
+
+                    if (vertexIndexToBoneIdWeight.Count > 0)
+                    {
+                        for (var bone = 0; bone < NumberOfBonesPerVertex; ++bone)
+                        {
+                            if (controlPointIndices16)
+                            {
+                                if (AllowUnsignedBlendIndices)
+                                    ((ushort*)(vbPointer + blendIndicesOffset))[bone] = (ushort)vertexIndexToBoneIdWeight[(int)i][bone].Item1;
+                                else
+                                    ((short*)(vbPointer + blendIndicesOffset))[bone] = (short)vertexIndexToBoneIdWeight[(int)i][bone].Item1;
+                            }
+                            else
+                            {
+                                if (AllowUnsignedBlendIndices)
+                                    ((byte*)(vbPointer + blendIndicesOffset))[bone] = (byte)vertexIndexToBoneIdWeight[(int)i][bone].Item1;
```

Comment:
This one seems to need a check for the bone index since the resulting list might end up having a size of less than 4